src/models/Review.js

```diff
@@ -1,16 +1,38 @@
 const mongoose = require('mongoose');
+const Product = require('./Product');
 
-const reviewSchema = new mongoose.Schema({
-    rating: {
-        type: Number,
-        min: 0,
-        max: 5,
-        default: 1,
+const reviewSchema = new mongoose.Schema(
+    {
+        rating: {
+            type: Number,
+            min: 0,
+            max: 5,
+            default: 1,
+        },
+        comment: {
+            type: String,
+        },
     },
-    comment: {
-        type: String,
+    { timestamps: true, versionKey: false },
+);
+
+// pre-hook on remove : when review is deleted then it should be removed from product
+// and it's contribution to product's average rating should also be removed
+reviewSchema.pre(
+    'deleteOne',
+    { document: true, query: false },
+    async function (next) {
+        const product = await Product.findOne({ reviews: this._id });
+        // console.log(`product : \n ${product}`);
+        product.rating = (product.rating * product.reviews.length - this.rating)
```

Comment:
Why are we updating each rating here, shouldn't we create a field for average rating, in that way we can maintain individual field, as well as average rating, or we can employ aggregation pipeline to calculate average rating just a throught, need to see tradeoffs here.